rust/private/rustc.bzl

```diff
@@ -217,6 +218,9 @@ def get_linker_and_args(ctx, attr, cc_toolchain, feature_configuration, rpaths):
             for linker_input in dep[CcInfo].linking_context.linker_inputs.to_list():
                 for flag in linker_input.user_link_flags:
                     user_link_flags.append(flag)
+
+    user_link_flags.extend(toolchain.stdlib_linkflags)
```

Comment:
There are 2 things that we need to take care of:

1) we need to make sure we pass `rust_toolchain.stdlib_linkflags` to `rustc` invocations doing linking. I think we're handling this use case already in this PR.
2) we need to put `rust_toolchain.stdlib_linkflags` to `CcInfo` provider. When we have a build like `cc_binary` depending on `rust_library`, linking action in `cc_binary` needs to contain all the `stdlib_linkflags`. I think we are not handling this use case currently.

I'm thinking how to best handle the 2). One approach is to provide `CcInfo` right from the `rust_toolchain`, and make sure we merge it in with the `CcInfo` of the current target. This will work, I'm not sure how good/bad idea it is to provide an additional provider from a toolchain, it's not done very often. Alternative is to add a rule that depends on the Rust toolchain, provides the `CcInfo`, and all other Rust rules implicitly depend on it to provide the `CcInfo`.

I lean towards the Rust toolchain providing `CcInfo`, but I'll try to nerd snipe @scentini into finding all the holes in that solution.

In any case though, could you add a test that:

* checks that `cc_binary` -> `rust_library` use case is handled
* checks that `stdlib_linkflags` appear after all `-l` flags for Rust crates
* checks that standalone `rust_binary` contains the flags?

And just to be sure, please make sure that all these tests still pass with the current implementation (without changes from this PR)

Reply:
> * checks that `stdlib_linkflags` appear after all `-l` flags for Rust crates

On my machine, I'm getting this with the current `rust_shared_library` and `rust_binary` tests:

```
"--codegen", "link-args=-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -B/usr/bin -pass-exit-codes -ldl -lpthread -lstdc++ -lm", 
"--extern=library_one=bazel-out/k8-fastbuild/bin/test/unit/platform_link_flags/liblibrary_one--1261616740.rlib", 
"--extern=library_two=bazel-out/k8-fastbuild/bin/test/unit/platform_link_flags/liblibrary_two--1109861386.rlib", 
"--extern=library_three=bazel-out/k8-fastbuild/bin/test/unit/platform_link_flags/liblibrary_three--1716324700.rlib", 
"-Ldependency=bazel-out/k8-fastbuild/bin/test/unit/platform_link_flags"
```

Reply:
Current status:

- [ ] checks that cc_binary -> rust_library use case is handled
- [ ] checks that stdlib_linkflags appear after all -l flags for Rust crates
- [x] checks that standalone rust_binary contains the flags?